src/matcha_ml/state/remote_state_manager.py

```diff
@@ -142,6 +153,11 @@ def is_state_provisioned(self) -> bool:
         if not self._configuration_file_exists():
             return False
 
+        if not self._resource_group_exists(
```

Comment:
Should we add some helpful message for users to know why the validation failed? Same for `_configuration_file_exists` and `_bucket_exists` or should IO be taken care by separate layer?

Reply:
I don't think users need to know about this. This is simply just whether we should create a new resource group and storage bucket or not based on if they exist.